pkg/aws/eni/types/types.go

```diff
@@ -128,6 +128,12 @@ type ENISpec struct {
 	//
 	// +kubebuilder:validation:Optional
 	UsePrimaryAddress *bool `json:"use-primary-address,omitempty"`
+
+	// DisablePrefixDelegation determines whether ENI prefix delegation should be
+	// disabled on this node.
+	//
+	// +kubebuilder:validation:Optional
+	DisablePrefixDelegation *bool `json:"disable-prefix-delegation,omitempty"`
```

Comment:
I think we should avoid the "disable" nomenclature vs "enable" because it avoids the potential double negative. In general, we are trying to avoid these potentially confusing options in Cilium as we used to have quite a few in the past (and some of them have remained).

Reply:
Agreed, but in this case, we explicitly wanted an opt-out flag instead of an opt-in one. The operator already has `aws-enable-prefix-delegation` and users only need to worry about the node one when some nodes need to excluded from PD. cc @gandro

Reply:
Yeah, to give more context here, this is named so that per-node PD is explicitly _opt-out_, and not _opt-in_. 

This simplifies the transition from when a user turns on global PD via the aforementioned operator flag. If the flag was opt-in, then one would have to enable it first on all `CiliumNode` CRDs, and then enable the flag globally. 
Thus the opt-out mechanic, because we assume that the majority of new nodes wants to use PD if it's globally enabled.

Having said that, we _could_ remove the `omitempty` here, and try to make use of tri-state logic (`nil`, `true`, `false`), where we assume that `DisablePrefixDelegation: nil` equals `true`. But I find that more confusing than the current semantics, where we assume that if the value is absent that it is `false`. 

So while I agree that the syntax is a bit inconsistent with our naming policy, I do believe that the semantics in this PR is the preferable one.

Reply:
Makes sense to  me.